processing/data_collection/gazette/spiders/pe_recife.py

```diff
@@ -0,0 +1,206 @@
+    The newest system presents gazettes in three ways:
+        - Pagination:
+            - Always available
+            - Document is a gazette's page
+            - Document is an image
+        - Exporting:
+            - Always available
+            - Document can be a gazette's page or the entire gazette
+            - Document is a PDF
+            - Takes a long time to export and sometimes fails to do it on large gazettes
+        - Attachment:
+            - Sometimes available
+            - When available, sometimes it's encrypted (so... not available)
+            - Document is the entire gazette
+            - Document is a PDF
+
+    Because "Exporting" is unreliable, "Attachment" is chosen whenever possible and
+    "Pagination" is the fallback.
+
+    There weren't found any indications in the responses' content to differentiate extra
+    from main editions and executive from legislative sections.
+
+Gazettes examples:
+    - `27/02/2018, main edition, 24 pages, with attachment, encrypted (.p7s)
+      <http://200.238.101.22/docreader/docreader.aspx?bib=R20180227&pasta=Fevereiro%5CDia%2027>`_
+    - `27/07/2019, main edition, 32 pages, with attachment, not encrypted
+      <http://200.238.101.22/docreader/docreader.aspx?bib=R20190727&pasta=Julho\Dia%2027>_
+    - `24/03/2020, extra edition, 16 pages, without attachment
+      <http://200.238.101.22/docreader/docreader.aspx?bib=R20200324&pasta=Marco%5CDia%2024>`_
+    - `07/04/2020, main edition, 8 pages, without attachment
+      <http://200.238.101.22/docreader/docreader.aspx?bib=R20200407&pasta=Abril\Dia%2007>`_
+"""
+
+import datetime as dt
+import re
+
+import dateparser
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class PeRecifeSpider(BaseGazetteSpider):
+    name = "pe_recife"
+    TERRITORY_ID = "2611606"
+    custom_settings = {"COOKIES_ENABLED": True, "AUTOTHROTTLE_ENABLED": True}
+
+    AVAILABLE_DATES_URL = "https://www.cepe.com.br/prefeituradiario/diarios.txt"
+    BASE_URL = (
+        "http://200.238.101.22/docreader/docreader.aspx?"
+        "bib=R{reversed_date}&pasta={month_full}%5CDia%20{day}"
+    )
+    PAGE_URL = (
+        "http://200.238.101.22/docreader/cache/{session_id}/I{gazette_page_id}-2"
+        "Alt={img_height}Lar={img_width}LargOri=003295AltOri=004299.JPG"
+    )
+
+    PAGE_PX_HEIGHT = 3295
+    PAGE_PX_WIDTH = 4299
+
+    months_full_names = {
+        "01": "Janeiro",
+        "02": "Fevereiro",
+        "03": "Marco",
+        "04": "Abril",
+        "05": "Maio",
+        "06": "Junho",
+        "07": "Julho",
+        "08": "Agosto",
+        "09": "Setembro",
+        "10": "Outubro",
+        "11": "Novembro",
+        "12": "Dezembro",
+    }
+
+    def start_requests(self):
+        yield from self.fetch_available_dates()
+
+    def fetch_available_dates(self):
+        yield scrapy.Request(self.AVAILABLE_DATES_URL, self.fetch_gazette_initial_page)
```

Comment:
Why not just move this line to the `start_requests` method?

Reply:
I thought it would be more descriptive. Particularly, I think that the `parse` or `start_requests` methods should always be "overriden" in favor of a more meaningful first step. Any ideas on how to make it better without implementing the second method?

Reply:
I think you can be more explicit just adding a comment. IMHO, another method is overkill.

Reply:
I ended up following you first suggestion :)